src/sidebar/services/test/groups-test.js

```diff
@@ -810,13 +794,33 @@ describe('groups', function () {
     });
   });
 
+  const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
+
   describe('automatic re-fetching', function () {
-    it('refetches groups when the logged-in user changes', () => {
-      service();
+    it('refetches groups when the logged-in user changes', async () => {
+      const svc = service();
 
-      return fakeRootScope.eventCallbacks[events.USER_CHANGED]().then(() => {
-        assert.calledOnce(fakeApi.groups.list);
-      });
+      // Load groups before profile fetch has completed.
+      fakeStore.hasFetchedProfile.returns(false);
+      await svc.load();
+
+      // Simulate initial fetch of profile finishing. This should be ignored.
+      fakeApi.groups.list.resetHistory();
+      fakeStore.hasFetchedProfile.returns(true);
+      fakeStore.profile.returns({ userid: 'acct:firstuser@hypothes.is' });
+      fakeStore.setState({});
+
+      await delay(1);
```

Comment:
Might you add a comment here about what this accomplishes?

Reply:
Yes. There are some async steps that happen before the group API fetch starts, so we need to wait before checking.